shared-libs/cht-datasource/src/local/libs/doc.ts

```diff
@@ -175,3 +175,12 @@ export const fetchAndFilterUuids = (
     limit
   );
 };
+
+/** @internal */
+export const createDoc = (db: PouchDB.Database) => async (data: Record<string, unknown>): Promise<Nullable<Doc>> => {
```

Comment:
```suggestion
export const createDoc = (db: PouchDB.Database<Doc>) => async (data: Record<string, unknown>): Promise<Nullable<Doc>> => {
```

if you change here then, the type cast in line 185 is not necessary

Reply:
Basically the object we pass to Pouch won't contain a `_rev` and `_id` right?